sdks/java/io/csv/src/main/java/org/apache/beam/sdk/io/csv/providers/CsvWriteTransformProvider.java

```diff
@@ -134,10 +135,15 @@ public PCollectionRowTuple expand(PCollectionRowTuple input) {
       if (configuration.getDelimiter() != null) {
         format = format.withDelimiter(configuration.getDelimiter().charAt(0));
       }
-      WriteFilesResult<?> result =
-          input
-              .get(INPUT_ROWS_TAG)
-              .apply(CsvIO.writeRows(configuration.getPath(), format).withSuffix(""));
+
+      // Preserve input windowing
+      CsvIO.Write<Row> writeTransform =
+          CsvIO.writeRows(configuration.getPath(), format).withSuffix("");
+      if (input.get(INPUT_ROWS_TAG).isBounded() == PCollection.IsBounded.UNBOUNDED) {
```

Comment:
This should probably be a function of the windowing (not being global), not of its boundedness.

Reply:
Good callout - I'm now wondering if there is a reason to not just omit a check entirely? The Javadoc for `withWindowedWrites()` states "Preserves windowing of input elements and writes them to files based on the element's window."

If this is the case, then wouldn't the windowing also preserve the global window? Or am I missing something?

Reply:
IIRC, this means shards will be named "/path/to/file-X-of-N-window_info.ext" rather than just bare "/path/to/file-X-of-N.ext." This is of course needed if there are multiple windows (to avoid collisions) but would be surprising in the global window (especially for batch users).

Reply:
Ah ok, I will go ahead and refactor that check then, thanks!